src/testrel.jl

```diff
@@ -36,7 +49,59 @@ set_context!(Context(RAI.load_config(fname="/Users/testing/.rai")))
 ```
 """
 function set_context!(new_context::Context)
-    return TEST_CONTEXT[] = new_context
+    return TEST_DEFAULTS[].context = new_context
+end
+
+"""
+    set_test_engine!(new_engine::Option{AbstractString})
+
+Set the engine to be used by the testing framework.
+
+If the engine is `nothing` and no `engine` is passed to a `@test_rel` call, the testing
+framework will take engines from the engine pool.
+
+# Examples
+
+To use a specific engine on tests:
+
+```
+set_test_engine!("my_engine")
+```
+
+To use the engine pool:
+
+```
+set_test_engine!(nothing)
+```
+"""
+function set_test_engine!(new_engine::Option{AbstractString})
+    return TEST_DEFAULTS[].engine = new_engine
+end
```

Comment:
This is a little redundant since there is already `add_test_engine!(name::String)` which adds the single listed engine to the pool. Given an empty pool the result is the same.

You are not the first to want to set a single default engine, though, so maybe it could map to
```
resize_test_engine_pool!(0)
add_test_engine!(name)
```

Reply:
I looked a bit more into the code, and I think there's still benefits in having this specific container for the "default engine". The solution I implemented here has 2 main advantages over mapping to `resize_test_engine_pool!` + `add_test_engine!`:

1. It is way more efficient, because `resize_test_engine_pool!` looks like a somewhat expensive operation, and `add_test_engine!` needs to actually call the RAI API to try to create the engine, and then detect a `409` to know the engine already exists. This is very heavyweight for just setting the name of the default engine.
2. It does not mess up with the pool, so it can be used temporarily. That is, I can have a pool with a few engines, then I set the default engine to run a few tests, then when I unset the default engine and my pool is still there with my original engines. I understand this is not a great use case, but I think it's better if we don't have those side effects when setting the default engine name.

What do you think about this? I am fine either way, I can rewrite to delegate like you suggest, but in my opinion having an explicit default engine configuration would be better.

Reply:
My objection is just the (minor) added complexity, but given it's an implementation detail rather than API I'm fine with it either way.